AllJoyn/Platform/BridgeHost/App.xaml.cs

```diff
@@ -0,0 +1,359 @@
+                if (command.Equals("AddDevice"))
+                {
+                    try
+                    {
+                        await AddDeviceToAllJoynAsync(onboardingData);
+                        response = "succeeded";
+                    }
+                    catch (Exception e)
+                    {
+                        response = e.Message;
+                    }
+                }
+                else if ((command != null) && command.Equals("healthcheck"))
+                {
+                    response = "healthy";
+                }
+            }
+            ValueSet valueSet = new ValueSet();
+            valueSet.Add("response", response);
+            await args.Request.SendResponseAsync(valueSet);
+            msgDeferral.Complete();
+        }
+
+        private void TaskInstance_Canceled(IBackgroundTaskInstance sender, BackgroundTaskCancellationReason reason)
+        {
+            appServiceDeferral.Complete();
+            System.Diagnostics.Debug.WriteLine("Task instance cancelled: {0}-{1}", sender.Task.Name, reason.ToString());
+        }
+
+        private  void AppServiceConnection_ServiceClosed(AppServiceConnection sender, AppServiceClosedEventArgs args)
+        {
+            appServiceDeferral.Complete();
+            System.Diagnostics.Debug.WriteLine("service close ConnectionForBackGroundProcess_ServiceClosed");
+        }
+
+        async Task OnboardRegisteredDevices()
+        {
+            // Cache the onboarding data and add the device to the bridge 
+            var appData = Windows.Storage.ApplicationData.Current.LocalSettings;
+
+            // Clear the cache the onboarded devices
+            var onboardedDevicesContainer = appData.CreateContainer(OnboardedDevicesContainerName, ApplicationDataCreateDisposition.Always);
+            onboardedDevicesContainer.Values.Clear();
+
+            foreach (var containerKey in appData.Containers.Keys)
+            {
+                var categoryContainer = appData.Containers[containerKey];
+                if(categoryContainer.Name.Equals(OnboardedDevicesContainerName)){
+                    continue; // Skip the temporary cache container.
+                }
+
+                foreach (var onboardingData in categoryContainer.Values.Values)
+                {
+                    try
+                    {
+                        var onboardingDataValueSet = AppDataCompositeValueToValueSet(onboardingData as ApplicationDataCompositeValue);
+                        await AddDeviceToAllJoynAsync(onboardingDataValueSet);
+
+                        // Store curently onboarded devices to AllJoyn bridge
+                        onboardedDevicesContainer.Values.Add(GetOnboardingId(onboardingDataValueSet), onboardingData as ApplicationDataCompositeValue);
+                    }
+                    catch (Exception e)
+                    {
+                        // Ignore
+                    }
```

Comment:
sure

---
In reply to: [75190428](https://github.com/tjaffri/msiot-samples/pull/9#discussion_r75190428) [](ancestors = 75190428)